### PR TITLE
[solution]: zh/medium-trim

### PR DESCRIPTION
**Pull request by SUtengda**
# Challenge [medium-trim]

## Checklist

- [x] Make sure that you are opening the PR with a single solution (one solution per PR)
- [x] The solution must have at least several references for the readers to dive in depth

### CODE REVIEW

File: zh/medium-trim.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+id: 108
+title: Trim
+lang: zh
+level: medium
+tags: template-literal
+---
+
+## 挑战
+
+实现`Trim<T>`，它是一个字符串类型，并返回一个新字符串，其中两端的空白符都已被删除。
```
Comment (charleserious) [inline zh/medium-trim.md:11]: I think we can not omit the "takes" here, which will be "它接收一个字符串类型"

File: zh/medium-trim.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+```
+
+## 解答
+
+跟[`TrimLeft<T>`](./medium-trimleft.md)几乎是相同的任务。
+我们这里用[模板字面量类型](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-1.html#template-literal-types) 在类型系统中对字符串进行建模
+
+现在有三种情况：空格在字符串的左边，空格在字符串的右边，以及字符串两边都不含空格。
+让我们先从左边有空格的情况开始建模。
+通过结合模板字面量类型和条件类型，我们可以推断对没有空格的其余部分推断。
```
Comment (charleserious) [inline zh/medium-trim.md:26]: duplicated infer translation here, and could be “我可以推断出不包含空字符串的其余部分”

**Review by charleserious — CHANGES_REQUESTED** [review]
Great work bro~
Issues check below~

**Comment by charleserious** [comment]
For the failing check, rebasing the main branch will solve the failure

File: zh/medium-trim.md (hunk 1 above)
Comment (SUtengda) [reply zh/medium-trim.md:11]: you are right, thanks

**Review by likui628 — APPROVED** [review]

**Review by ghaiklor — APPROVED** [review]

**Comment by ghaiklor** [comment]
Forcing the merge, seems like comments from @charleserious were resolved, and we got another approval.

**Pull request merged**